pages/biosero_plate_state.vue

```diff
@@ -0,0 +1,223 @@
+<template>
+  <b-container style="max-width: 1400px">
+    <h1 class="mt-3">Biosero Plate state</h1>
+    <p class="lead">Find the state of a source or destination plate</p>
+    <b-form-group
+      id="plate-barcode"
+      description="Scan in a destination or source plate barcode to search for its state in Biosero"
+      label="Plate barcode:"
+      label-for="plate-barcode-field"
+    >
+      <b-form-input
+        id="plate-barcode-field"
+        ref="plate-barcode-field"
+        v-model="barcode"
+        type="search"
+        trim
+      ></b-form-input>
+    </b-form-group>
+
+    <p>
+      <b-button id="searchPlates" variant="info" @click="findPlate">Search</b-button>
+    </p>
+
+    <Alert ref="alert"></Alert>
+
+    <b-card id="plate-summary" title="Plate Summary">
+      <div v-if="plate.source">
+        <b-card-text
+          ><b>Source plate barcode:</b> {{ lastPlateBarcode }}</b-card-text
+        >
+        <b-card-text style="color: green"
+          >Total number of picked wells: {{ calculateSourceWells[0] }}</b-card-text
+        >
+        <b-card-text style="color: darkorange"
+          >Total number of unpicked wells: {{ calculateSourceWells[1] }}</b-card-text
+        >
+        <b-card-text>Total number of empty wells: {{ calculateSourceWells[2] }}</b-card-text>
+      </div>
+
+      <div v-if="plate.destination">
+        <b-card-text
+          ><b>Destination plate barcode:</b> {{ lastPlateBarcode }}</b-card-text
+        >
+        <b-card-text style="color: green"
+          >Total number of picked wells: {{ calculateDestinationWells[0] }}</b-card-text
+        >
+        <b-card-text style="color: gray"
+          >Total number of control wells: {{ calculateDestinationWells[1] }}</b-card-text
+        >
+        <b-card-text>Total number of empty wells: {{ calculateDestinationWells[2] }}</b-card-text>
+      </div>
+
+      <b-form-group id="plate-filter-group" label="Filters:" label-for="plateFilter">
+        <b-form-select
+          id="plate-filter"
+          v-model="filter"
+          style="width: 250px"
+          :options="plateFilterOptions"
+          :disabled="!plate.source && !plate.destination"
+        />
+      </b-form-group>
+    </b-card>
+
+    <div id="plate-container">
+      <b-table id="plate-table" :fields="plateFields" :items="plateItems" fixed bordered></b-table>
+    </div>
+  </b-container>
+</template>
+
+<script>
+import Alert from '@/components/AlertDialog'
+import lighthouseBiosero from '@/modules/lighthouse_service_biosero'
+
+export default {
+  name: 'BioseroPlateState',
+  components: {
+    Alert,
+  },
+  data() {
+    return {
+      barcode: '',
+      lastPlateBarcode: '',
+      plate: { source: false, destination: false },
+      filter: 'source_barcode',
+      plateFields: [
+        { key: 'row', label: '', isRowHeader: true },
+        '1',
+        '2',
+        '3',
```

Comment:
Something doesn't feel right about this.  You've got a hash as the first entry in this array, but that feels like it shouldn't be part of the array itself as it looks like metadata for this whole array.  I think later this array is iterated over which confused me as to how this works.  Should this data be structured differently?  Something like:

```
plateFields: [
  {
    key: 'row',
    label: '',
    isRowHeader: true,
    values: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12]
  }
]
```

Reply:
I agree it does look a little weird but the first hash is interpreted as a value with metadata by bootstrap so in itself it is a value. docs here which will explain it better than I do: 
https://bootstrap-vue.org/docs/components/table

Reply:
Thanks!  Today I learned something new.